pkg/migrations/sql/idempotent/telemetry.sql

```diff
@@ -0,0 +1,94 @@
+-- promscale_telemetry_housekeeping() does telemetry housekeeping stuff, which includes
+-- searching the table once an hour, looking for stale promscales, if found,
+-- adding their values into the counter_reset row, and then cleaning up the stale
+-- promscale instances.
+-- It is concurrency safe, since it takes lock on the promscale_instance_information table,
+-- making sure that at one time, only one housekeeping is being done.
+-- 
+-- It returns TRUE if last run was beyond an hour, otherwise FALSE.
+CREATE OR REPLACE FUNCTION SCHEMA_PS_CATALOG.promscale_telemetry_housekeeping()
+RETURNS BOOLEAN AS
+$$
+    DECLARE
+        is_last_run_beyond_hour BOOLEAN := FALSE;
+    BEGIN
+        LOCK TABLE SCHEMA_PS_CATALOG.promscale_instance_information IN ACCESS EXCLUSIVE MODE NOWAIT; -- Do not wait for the lock as some promscale is already cleaning up the stuff.
+
+        -- Check if the last run was beyond an hour.
+        IF (SELECT count(*) > 0 FROM SCHEMA_PS_CATALOG.promscale_instance_information
+            WHERE
+                is_counter_reset_row = TRUE
+            AND
+                current_timestamp - last_updated < INTERVAL '1 HOUR') THEN
+            -- Some Promscale did the housekeeping work within the expected interval. Hence, nothing to do, so exit.
+            RETURN is_last_run_beyond_hour;
+        END IF;
+
+        -- Last run was beyond an hour.
+        is_last_run_beyond_hour := TRUE;
+
+        -- The last_update of counter_reset row corresponds to the last run of housekeeping, and not its actual meaning, which was
+        -- when the counter_reset row was updated. This is done so that there can be place to store the last executing time,
+        -- allowing all Promscales to be housekeepers, instead of just one, thereby decreasing the async complexity.
+        -- 
+        -- Hence, update the last_update of counter reset row since now is its recent most run.
+        UPDATE SCHEMA_PS_CATALOG.promscale_instance_information SET last_updated = current_timestamp WHERE is_counter_reset_row = TRUE;
```

Comment:
If there ARE stale rows present, we end up running 2 update statements on the counter reset row. I suggest rearranging this a bit so that we only ever issue one update against the row in this function.

Reply:
Sorry, the last_update of counter_reset row is not the last_update of that row, rather the last time when the table was touched for housekeeping.

> NOTE FOR REVIEWERS: Now, all promscales act as housekeepers. If the last housekeeping was done beyond an hour, then do the checking of stale promscales, counter reset increment, and evaluation of SQL-based telemetry stats (in the coming PRs). The state of last housekeeping done, is saved in the last_update attribute of counter reset row, than saving somewhere else, meaning counter reset row's last_updated is no longer the actual meaning.